src/app/campaign/flight/flight-zones-form.component.ts

```diff
@@ -45,7 +45,13 @@ export const validateMp3 = (value: string): { [key: string]: any } | null => {
         </mat-form-field>
         <mat-form-field appearance="outline">
           <mat-label>Creative Url</mat-label>
-          <input matInput formControlName="url" autocomplete="off" placeholder="Leave blank for a 0-second MP3" />
+          <input
+            matInput
+            [errorStateMatcher]="matcher"
```

Comment:
So what's the difference between this and the default behavior?  As far as i can tell, this makes the error message appear right away rather than on-blur.  Just making sure I'm on the right track here.

(And I remember angrily trying to get that error message to show before-blur when working on the zones input ... good to know there's a way to fix that).

Reply:
The default behavior is to show errors when invalid and dirty. I was seeing invalid values in the field that weren't highlighted, and I couldn't see right away why the form was invalid. (I think the value may have carried over from another flight with that carry over bug, but it bothered the hecc out of me so I made it show the validation error.)